.github/workflows/build_push_service.yml

```diff
@@ -6,20 +6,169 @@ name: sc-app-build-push
+      - name: Read version file
+        # The version to be used for tagging the images is read out of the version file
+        id: read_version
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./version
+      - name: Set env AVCHECK_VERSION
+        run: |
+          echo "Version of antivirus_service is ${{ steps.read_version.outputs.content }}"
+          echo "AVCHECK_VERSION=${{ steps.read_version.outputs.content }}" >> $GITHUB_ENV
+      - name: RegEx JIRA ticket ID
+        uses: actions-ecosystem/action-regex-match@v2
+        id: jira
+        with:
+          text: ${{ env.BRANCH_NAME }}
+          regex: '^hotfix|feature\/([A-Z]+-[0-9]+)-[a-zA-Z_]+$'
+          flags: gm
+      - name: Set env var JIRA_TicketID
+        run: |
+          echo "JIRA_TicketID is ${{ steps.jira.outputs.group1 }}"
+          echo JIRA_TicketID=${{ steps.jira.outputs.group1 }} >> $GITHUB_ENV
+      # Assemble the Dockertag for the build image based on the type of the branch
+      - name: Find JIRA ticket BRANCH_PREFIX
+        uses: actions-ecosystem/action-regex-match@v2
+        id: branch_prefix
+        with:
+          text: ${{ env.BRANCH_NAME }}
+          regex: '^([a-zA-Z]+)\/[A-Z]+-[0-9]+-[a-zA-Z_]+$'
+          flags: gm
+      - name: Set env var JIRA BRANCH_PREFIX
+        run: |
+          echo "BRANCH_PREFIX is ${{ steps.branch_prefix.outputs.group1 }}"
+          echo BRANCH_PREFIX=${{ steps.branch_prefix.outputs.group1 }} >> $GITHUB_ENV
+      # Same convention as implemented OPS-1664
+      # develop -> just develop_latest
+      - name: Assemble Dockertag for develop
+        if: env.BRANCH_PREFIX == 'develop'
+        run: echo DOCKER_TAG_POSTFIX=develop_latest >> $GITHUB_ENV
+      # feature -> feature_<JIRA ID>_latest
+      - name: Assemble Dockertag for feature
+        if: env.BRANCH_PREFIX == 'feature'
+        run: echo "DOCKER_TAG_POSTFIX=feature_${{env.JIRA_TicketID}}_latest" >> $GITHUB_ENV
+      # release -> release_v<AVCHECK_VERSION>_latest, release_v<AVCHECK_VERSION>_GITSHA
+      - name: Assemble Dockertag for release
+        if: env.BRANCH_PREFIX == 'release'
+        run: |
+          echo "DOCKER_TAG_POSTFIX=release_${{env.AVCHECK_VERSION}}_latest" >> $GITHUB_ENV
+          echo "DOCKER_TAG_SHA=release_${{env.AVCHECK_VERSION}}_${{env.GITHUB_SHA}}" >> $GITHUB_ENV
+      # hotfix -> hotfix_<JIRA ID>_latest
+      - name: Assemble Dockertag for hotfix
+        if: env.BRANCH_PREFIX == 'hotfix'
+        run: |
+          echo "DOCKER_TAG_POSTFIX=hotfix_${{env.JIRA_TicketID}}_latest" >> $GITHUB_ENV
+      # master -> master_v<AVCHECK_VERSION>_latest, master_v<AVCHECK_VERSION>_GITSHA
+      - name: Assemble Dockertag for master
+        if: env.BRANCH_PREFIX == 'master'
+        run: |
+          echo "DOCKER_TAG_POSTFIX=master_${{env.AVCHECK_VERSION}}_latest" >> $GITHUB_ENV
+          echo "DOCKER_TAG_SHA=master_${{env.AVCHECK_VERSION}}_${{env.GITHUB_SHA}}" >> $GITHUB_ENV
```

Comment:
Too much special cases, why was they not designed as separate workflows?

Reply:
See epic, to meet naming convention it is necessary.

Reply:
See epic, to meet naming convention it is necessary. We will build all branches with one Workflow.
One thing was that actions have very little capability's on conditions. It doesn't look smart but now it works.